File: generate.kt
import java.io.File

data class Entry(val word: String) {
    val letters = word.toSet()
}

fun loadWords() = File("words").readLines().map { Entry(it.trim()) }.filter { it.word.length >= 4 }

val words = loadWords()
val panagrams = words.filter { it.letters.size == 7 }

data class Game(val central: Char, val others: List<Char>, val matching: List<String>) {
    init {
        if (others.size != 6) {
            throw Exception("Expecting 6 other characters; got $others")
        }
        if (others.toSet().size != 6) {
            throw Exception("Duplicate characters: $others")
        }
    }

    companion object {
        fun generate(): Game {
            val panagram = panagrams.random()
            val central = panagram.letters.random()

            val matching = words.filter {
                panagram.letters.containsAll(it.letters) && it.letters.contains(central)
            }
            if (!matching.contains(panagram)) {
                throw Exception("Panagram ($panagram) not found in matching words")
            }

            val allPanagrams = matching.filter { it.letters.size == 7 }
            val matchingList = allPanagrams.sortedBy {it.word} + (matching - allPanagrams).sortedBy{it.word}

            return Game(central, (panagram.letters - setOf(central)).toList(), matchingList.map { it.word })
        }
    }
}

fun main() {
    val game = Game.generate()
    println(game)
}

File: generate.py
import random
from dataclasses import dataclass
from typing import NamedTuple, FrozenSet, List


class Entry(NamedTuple):
    word: str
    letters: FrozenSet[str]


@dataclass
class Game:
    central: str
    others: str
    matching: List[str]

    def __post_init__(self):
        assert len(self.central) == 1
        assert len(self.others) == 6
        assert len(set(self.others)) == 6

    @classmethod
    def generate(cls):
        panagram = random.choice(panagrams)
        central = random.choice(list(panagram.letters))

        matching = {
            entry
            for entry in words
            if entry.letters <= panagram.letters
            and central in entry.letters
            and len(entry.word) >= 4
        }
        assert panagram in matching
        all_panagrams = {word for word in matching if len(set(word)) == 7}
        matching_list = sorted(all_panagrams) + sorted(matching - all_panagrams)

        return cls(
            central, list(panagram.letters - {central}), [w.word for w in matching_list]
        )


def load_words():
    with open("words") as f:
        words = [word.strip() for word in f]

    return [Entry(word, frozenset(word)) for word in words]


words = load_words()
panagrams = [entry for entry in words if len(entry.letters) == 7]
